hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/SqlAnalyzedJobSuspensionIsForbiddenTest.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2023 Hazelcast Inc.
+ *
+ * Licensed under the Hazelcast Community License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://hazelcast.com/hazelcast-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.sql.impl;
+
+import com.hazelcast.cluster.ClusterState;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.jet.Job;
+import com.hazelcast.jet.SimpleTestInClusterSupport;
+import com.hazelcast.jet.core.JetTestSupport;
+import com.hazelcast.jet.core.JobStatus;
+import com.hazelcast.test.HazelcastSerialClassRunner;
+import com.hazelcast.test.annotation.ParallelJVMTest;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import java.util.Objects;
+import java.util.concurrent.CancellationException;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.junit.Assert.assertNotNull;
+
+/**
+ * Note: We prepared a separate test suite to prevent tests to be flaky
+ *  because {@link AnalyzeStatementTest} inherits {@link SimpleTestInClusterSupport},
```

Comment:
do you expect any serious problems with these tests? They are rather simple now and not prone to race conditions

Reply:
Yes, at least I want to prevent any problems corresponded with usage of `SimpleTestInClusterSupport` for multiple streaming queries tests.

Reply:
`SimpleTestInClusterSupport` causes problems only when the tests cause timeouts. Or do you mean different problems?

Reply:
I had a problem, when you run all the tests, some may be flaky in the suite with `SimpleTestInClusterSupport`. To prevent this, and any possible flakyness in the future, I separate all tests with streaming queries and do not share any instanses.

Reply:
could you be more specific if `SqlAnalyzedJobSuspensionIsForbiddenTest` caused or suffered from flakiness? If not, then I do not see any reason to give up `SimpleTestInClusterSupport` for it. SimpleTestInClusterSupport tests are faster.

Reply:
Again - I had a problem with tests, which I recently moved to `SqlAnalyzedJobSuspensionIsForbiddenTest`. When I ran all tests in batch, some of them just hanged. I experienced such behavior before, and the solution was to separate these tests and control members termination manually.

Reply:
maybe they hanged because of https://github.com/hazelcast/hazelcast/pull/25939#discussion_r1394446645?

Reply:
ultimate fix in #26007